src/main.rs

```diff
@@ -41,23 +41,45 @@ fn main() {
 
     loop {
         let cur_action = action_stack.current();
-        let user_selection = UserPrompt::from_str(&user_data.get_user_result(cur_action)).unwrap();
+
+        let user_selection = match user_data.get_user_prompt(cur_action) {
+            Ok(prompt) => prompt,
+            Err(err) => {
+                action_stack.push(UserAction::Error(format!(
+                    "could not process prompt: {}",
+                    err
+                )));
+                continue;
+            }
+        };
+
         match user_selection {
             UserPrompt::Back => {
                 action_stack.pop();
             }
             UserPrompt::Exit => break,
             UserPrompt::StartOver => action_stack.clear(),
```

Comment:
`user_data` needs to be cleared here so that it doesn't carry over between iterations. Something like:

```rust
            UserPrompt::StartOver => {
                action_stack.clear();
                user_data = UserMetadata::default();
            }

```
Ought to do the trick.